yarn-project/pxe/src/contract_function_simulator/execution_data_provider.ts

```diff
@@ -214,13 +214,36 @@ export interface ExecutionDataProvider {
   assertCompatibleOracleVersion(version: number): void;
 
   /**
-   * Returns the next app tag for a given sender and recipient pair.
+   * Calculates the directional app tagging secret for a given contract, sender and recipient.
    * @param contractAddress - The contract address to silo the secret for
    * @param sender - The address sending the note
    * @param recipient - The address receiving the note
+   * @returns The directional app tagging secret
+   */
+  calculateDirectionalAppTaggingSecret(
+    contractAddress: AztecAddress,
+    sender: AztecAddress,
+    recipient: AztecAddress,
+  ): Promise<Fr>;
+
+  /**
+   * Updates the local index of the shared tagging secret of a (sender, recipient, contract) tuple if a log with
+   * a larger index is found from the node.
+   * @param directionalAppTaggingSecret - The secret that's unique for (sender, recipient, contract) tuple while
+   * direction of sender -> recipient matters.
+   * @param contractAddress - The address of the contract that the logs are tagged for. Needs to be provided to store
+   * because the function performs second round of siloing which is necessary because kernels do it as well (they silo
+   * first field of the private log which corresponds to the tag).
+   */
+  syncTaggedLogsAsSender(directionalAppTaggingSecret: Fr, contractAddress: AztecAddress): Promise<void>;
```

Comment:
I needed to expose this because now the privateGetNextAppTagAsSender calls this.